### PR TITLE
cli: option --watcher

### PR DESCRIPTION
**Pull request by chrmod**
Implements another part of https://github.com/broccolijs/broccoli/issues/331

### CODE REVIEW

**Comment by chrmod** [comment]
@stefanpenner @rwjblue please take a look. the tests could probably be less verbose so if you have suggestion on how to improve please share.

File: lib/cli.js [hunk 1]
```diff
@@ -121,6 +123,25 @@ function getWatcher(isWatching) {
...
+  const watcher = options.watcher;
+
+  if (watcher && ['polling', 'watchman', 'node', 'events'].indexOf(watcher) === -1) {
+    throw new Error(`Unknown watcher type --watcher=[polling|watchman|node] but was: ${watcher}`);
+  }
+
+  return {
+    saneOptions: {
+      poll: watcher === 'polling',
+      watchman: watcher === 'watchman' || watcher === 'events' || !watcher,
```
Comment (stefanpenner) [inline lib/cli.js]: I think we may want to consider dropping `events` as `watchman` is both clear and concise and already covers that one.

Comment (stefanpenner) [reply lib/cli.js]: If there is no watcher specified,  it likely isn't safe to just assume `watchman` (that would also be backwards incompatible).

We should likely either default to `node` or use `execSync` to see if the `watchman version` returns some minimum version.

Comment (chrmod) [reply lib/cli.js]: seems also ci detected the problem. fixing it now

**Comment by chrmod** [comment]
@stefanpenner I've changed default back to `node` and tests are passing again

**Comment by stefanpenner** [comment]
@chrmod LGTM.

* I think future PR could try to autodetect watchman and prefer that
* I plan to get a broccoli release out this weekend, so we can get your improvements in the wild

**Comment by chrmod** [comment]
@stefanpenner great, thanks for pushing it forward

**Pull request merged**